engine/execution/ingestion/engine.go

```diff
@@ -562,7 +561,11 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 
 	isExecutedBlockSealed := executableBlock.Block.Header.Height <= lastSealed.Height
 	broadcasted := false
-	if !isExecutedBlockSealed && e.staker.AmIStakedAt(executableBlock.ID()) {
+	stakedAtBlock, err := e.checkStakedAtBlock(executableBlock.ID())
+	if err != nil {
+		e.log.Fatal().Err(err).Msg("could check staking status")
```

Comment:
I'm not sure about the flow of how executable blocks arrive at this function, but this means we'll crash for any inputs we haven't incorporated into the protocol state. Is that OK?

Reply:
If we are asked to executed a block, then the block must have been incorporated to protocol state. So the block must be found in protocol state. Meaning any error here would be bug.